#!/usr/bin/env python
PACKAGE = "base_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_area", 				int_t,    0, "An Integer parameter", 100,	0,10000)
#blue
gen.add("blue_H_low", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("blue_H_high", 			int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("blue_S_low",  			int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("blue_S_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("blue_V_low",  			int_t,    0, "An Integer parameter", 230,	0,255)
gen.add("blue_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
#red
gen.add("red_H_low1", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("red_H_high1", 			int_t,    0, "An Integer parameter", 30,	0,179)
gen.add("red_H_low2", 			int_t,    0, "An Integer parameter", 178,	0,179)
gen.add("red_H_high2", 			int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("red_S_low",  			int_t,    0, "An Integer parameter", 20,	0,255)
gen.add("red_S_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("red_V_low",  			int_t,    0, "An Integer parameter", 160,	0,255)
gen.add("red_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
#black
# gen.add("black_H_low", 			int_t,    0, "An Integer parameter", 0,		0,179)
# gen.add("black_H_high", 		int_t,    0, "An Integer parameter", 255,	0,179)
# gen.add("black_S_low",  		int_t,    0, "An Integer parameter", 200,	0,255)
# gen.add("black_S_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
# gen.add("black_V_low", 			int_t,    0, "An Integer parameter", 0,		0,255)
# gen.add("black_V_high",			int_t,    0, "An Integer parameter", 60,	0,255)

exit(gen.generate(PACKAGE, "base_vision", "armor_color"))